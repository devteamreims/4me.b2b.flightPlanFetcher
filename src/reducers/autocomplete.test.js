import R from 'ramda';
import MockDate from 'mockdate';

import {
  markForAutocomplete,
} from '../actions/autocomplete';

import {
  invalidIfplId,
} from '../actions/flightKeys';

import reducer from './autocomplete';



describe('autocomplete', () => {
  describe('markForAutocomplete', () => {
    test('should reject invalid arguments', () => {
      expect(() => markForAutocomplete()).toThrowError(/argument/i);
      expect(() => markForAutocomplete({})).toThrowError(/argument/i);
      expect(() => markForAutocomplete({a: 'test'})).toThrowError(/argument/i);
      expect(() => markForAutocomplete('string')).toThrowError(/argument/i);
      expect(() => markForAutocomplete([{}, 2])).toThrowError(/argument/i);
    });

    test('should accept valid ifplIds list', () => {
      expect(() => markForAutocomplete(['ifplId1', 'iplfId2'])).not.toThrow();
    });
  });
  describe('reducer', () => {
    test('should have an initial state', () => {
      expect(reducer(undefined, {type: '@@INIT'})).toEqual([]);
    });

    test('should mark specific keys are part of the autocomplete list', () => {
      const ifplIds = ['AA0', 'AA1', 'AA2'];
      const action = markForAutocomplete(ifplIds);
      expect(reducer(undefined, action)).toMatchSnapshot();
    });

    test('should discard old ifplIds marked for autocomplete', () => {
      const ifplIds = ['AA0', 'AA1', 'AA2'];
      let state;
      state = reducer(state, markForAutocomplete(ifplIds));
      state = reducer(state, markForAutocomplete(['AA1', 'AA3']));
      expect(state).toMatchSnapshot();
    });

    test('should remove keys when invalidated', () => {
      const ifplIds = ['AA0', 'AA1', 'AA2'];
      let state;
      state = reducer(state, markForAutocomplete(ifplIds));
      state = reducer(state, invalidIfplId('AA2'));
      expect(state).not.toContain('AA2');
    });
  });
});
