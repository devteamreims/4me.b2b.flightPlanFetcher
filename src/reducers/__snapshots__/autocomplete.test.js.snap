exports[`autocomplete reducer should discard old ifplIds marked for autocomplete 1`] = `
Array [
  "AA1",
  "AA3",
]
`;

exports[`autocomplete reducer should mark specific keys are part of the autocomplete list 1`] = `
Array [
  "AA0",
  "AA1",
  "AA2",
]
`;
